src/Invictus.Testing.Tests.Integration/LogicAppsProviderTests.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Invictus.Testing.Model;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Invictus.Testing.Tests.Integration
+{
+    public class LogicAppsProviderTests : IntegrationTest
```

Comment:
Maybe add some explicit test cases where:
- Criteria is not matching
- That uses LocatedAt overload without logger